share/eds/ods_test.go

```diff
@@ -0,0 +1,52 @@
+package eds
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/ipld/go-car"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+// TestODSReader ensures that the reader returned from DSReader is capable of reading the CAR header and ODS.
+func TestODSReader(t *testing.T) {
```

Comment:
it is helpful to add comments that explain the steps of the test so that reviewers and future developers can more easily verify that the test is doing what we expect it to be doing.

Reply:
For example, we should either explicitly test the `Read` method or explain how it is being tested with this test.

Reply:
I've added more comments for test to make it easier to follow steps.